### ISSUE TITLE
X Error of failed request:  BadRRCrtc (invalid Crtc parameter)

### ISSUE THREAD

**Issue by plazmakeks** [glfw/glfw#1147]
Hi there,

i recently updated my glfw version and face a problem now. I'm on the master branch (commit hash: 9e6c0c747be838d1f3dc38c2924a47a42416c081). 

Thats my test program:

```c
#include <glad/glad.h>
#include <GLFW/glfw3.h>

#include <stdio.h>

void onGLFWEerror(int _error, const char* _desc) {
	printf("GLFW error: %d, desc: %s \n", _error, _desc);
}

int main() {
	
	int major = 0, minor = 0, rev = 0;
	glfwGetVersion(&major, &minor, &rev);
	printf("using GLFW %d.%d.%d\n", major, minor, rev);

	glfwInit();

	glfwSetErrorCallback(onGLFWEerror);

	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
	glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
	glfwWindowHint(GLFW_RESIZABLE, GLFW_TRUE);
	glfwWindowHint(GLFW_AUTO_ICONIFY, GLFW_TRUE);

	int countMonitors;
	GLFWmonitor** monitors = glfwGetMonitors(&countMonitors);
	printf("# monitors: %d\n", countMonitors);
	auto monitor = monitors[0];

	int xpos, ypos;
	int widthMM, heightMM;
	int cVideoModes;

	auto name = glfwGetMonitorName(monitor);
	printf("---MONITOR INFO---\n");
	if(monitor == NULL) {
		printf("monitor is NULL\n");
	}
	printf("name: %s\n", name);
	glfwGetMonitorPos(monitor, &xpos, &ypos);
	printf("xpos: %d, ypos: %d\n", xpos, ypos);
	glfwGetMonitorPhysicalSize(monitor, &widthMM, &heightMM);
	printf("widthMM: %d, heightMM: %d\n", widthMM, heightMM);
	auto videoModes = glfwGetVideoModes(monitor, &cVideoModes);

	return 0;
}
```
It works on my desktop machine but crashes with this output
```
using GLFW 3.3.0
# monitors: 1
---MONITOR INFO---
name: Display
X Error of failed request:  BadRRCrtc (invalid Crtc parameter)
  Major opcode of failed request:  140 (RANDR)
  Minor opcode of failed request:  20 (RRGetCrtcInfo)
  Crtc id in failed request: 0x0
  Serial number of failed request:  98
  Current serial number in output stream:  98
```
on a machine which has this xrandr setup:
```
# xrandr -q
Screen 0: minimum 8 x 8, current 8 x 8, maximum 32767 x 32767
DP1 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
HDMI2 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
```
Is it a bug or can i solve it on my own?

It appeared yesterday firstly after updating my glfw version from git to the named version.

Best,
Henning

**Comment by elmindreda** [comment]
> Is it a bug or can i solve it on my own?

It's a bug. My bug, to be exact, and a regression since the last release.

**Comment by plazmakeks** [comment]
> It's a bug. My bug, to be exact, and a regression since the last release.

Thanks for your quick response. Please let me know when it's solved. 

Thanks,
Henning

**Comment by elmindreda** [comment]
This should be fixed now with 30489c5aa1392f698e904c6aced1d7a486634c40. Note that your code assumes at least one monitor, but you won't get any on that remote system.

**Comment by antylama** [comment]
Hello, I'm trying to use glfw related library (raylib-go) on machine without monitor (VLC vino server). I have the same errors, like in this thread. After updating external dependency of raylib-go (glfw with fix 30489c5), I have another problem - raylib really needs active monitor (it's hard to fix it).

Another way, examples of glfw, take care of is real monitor plug or not, but for example: "particles.c" have optional argument _-f_ with one not work (open program in fullscreen mode) on VLC only connection.

Set `monitorBroken` to `GLFW_TRUE`, "fix" this problems for me, but maybe it's not correct:

```diff
--- a/src/x11_monitor.c
+++ b/src/x11_monitor.c
@@ -203,7 +203,7 @@ void _glfwPollMonitorsX11(void)

         free(disconnected);
     }
-    else
+    if (!_glfw.monitorCount)
     {
         const int widthMM = DisplayWidthMM(_glfw.x11.display, _glfw.x11.screen);
         const int heightMM = DisplayHeightMM(_glfw.x11.display, _glfw.x11.screen);
@@ -211,6 +211,7 @@ void _glfwPollMonitorsX11(void)
         _glfwInputMonitor(_glfwAllocMonitor("Display", widthMM, heightMM),
                           GLFW_CONNECTED,
                           _GLFW_INSERT_FIRST);
+        _glfw.x11.randr.monitorBroken = GLFW_TRUE;
     }
 }
```
